##################################################################################################################
# Check and compare % of mito. localization in genes above threshold (p<0.01 or 0.05 & ES>|0.2|) and in all genes of DEA #
##################################################################################################################


# required libraries
require(tidyverse)
library(magrittr)
library(rvest)
library(caroline)
require(data.table)
require(ggsci)
require(ggpubr)
require(viridis)


##################################
# create paths
system("mkdir Figs/DEA/")
fig_path <- c("Figs/DEA/")
fig_path_pdf <- c("Figs/v8/pdf/")


hmitocarta = "Data/gene_sets/MitoCarta3.0/MitoCarta_highconfidence.csv"
hmitocarta_tbl = fread(hmitocarta, header = T)

allmitocarta = "Data/gene_sets/MitoCarta3.0/MitoCarta_all.csv"
allmitocarta_tbl = fread(allmitocarta, header = T)

##################################
# import DEA lmm output files

# rna_DEA1 <- fread("interimData/DEA/diff_exp_rnaseq_pathwayact_all_annotout.txt")
rna_DEA <- fread("interimData/DEAlmm/diff_exp_data_rnaseq_pathwayact_all.txt")
rna_DEA_cond <- fread("interimData/DEA/diff_exp_rnaseq_pathwayact_conditional_annotout.txt")

# prot_DEA1 <- fread("interimData/DEA/diff_exp_prot_pathwayact_all_annotout.txt")
prot_DEA <- fread("interimData/DEAlmm/diff_exp_data_prot_pathwayact_all.txt")
prot_DEA_cond <- fread("interimData/DEA/diff_exp_prot_pathwayact_conditional_annotout.txt")

### POSITIVE beta value --> RNA is up in samples with high pathway activity
### NEGATIVE beta value --> RNA is down in samples with high pathway activity
### samples with low pathway activity are mostly MMUT deficient samples
### betas and pval according to DL's fast_lmm model


# create gene lists of interest
TCA <- "Data/gene_sets/masterTCA_keggPlusmanual_circos.csv" #with master_TCA file generated by script
tca_tbl <- fread(TCA, header = TRUE)[,ENSG]
tca_tbl <- as.data.table(tca_tbl)
colnames(tca_tbl) <- "ENSG"
hgnc_r <- fread("Data/gene_sets/ENSG_HGNC_list_r.csv")
setnames(hgnc_r, c("ENSG", "HGNC"))
tca_genes_tbl <- inner_join(tca_tbl, hgnc_r, by="ENSG")
tcagenes_new <- tca_genes_tbl$HGNC

tca_genes_dt <- fread("Data/gene_sets/tca_geneset_manual_wHGNC.csv", header = TRUE)
tca_genes <- tca_genes_dt$hgnc_symbol
# genes_of_interest <- c("MMUT", "OGDH", "GLUD1", "GLUD2", "PDK4", "GOT2", "SUCLA2", "DLD", "GOT1","DLST", "CS")
genes_of_interest_rna <- c("MMUT", "PDK4",  "SUCLA2")
genes_of_interest_prot <- c("MMUT", "OGDH", "GLUD1", "SUCLA2")

# genes_of_interest <- c("MMUT", "PDK4", "SUCLA2") # "DLD", "GOT1","DLST", "CS",

# prep RNA table
for(i in 1:length(tcagenes_new)) {
  rna_DEA[which(rna_DEA$gname %in% tcagenes_new[i]), label := as.character(tcagenes_new[i])]
}

for(i in 1:length(genes_of_interest_rna)) {
  rna_DEA[which(rna_DEA$gname %in% genes_of_interest_rna[i]), label1 := as.character(genes_of_interest_rna[i])]
}

rna_DEA[pval<0.01, label3 := gname]
rna_DEA[!is.na(label), ]
rna_DEA[!is.na(label1), ]
rna_DEA[!is.na(label3), ]


# prep protein table
for(i in 1:length(tcagenes_new)) {
  prot_DEA[which(prot_DEA$gname %in% tcagenes_new[i]), label := as.character(tcagenes_new[i])]
}

for(i in 1:length(genes_of_interest_prot)) {
  prot_DEA[which(prot_DEA$gname %in% genes_of_interest_prot[i]), label1 := as.character(genes_of_interest_prot[i])]
}

prot_DEA[pval<0.01, label3 := gname]
prot_DEA[!is.na(label), ]
prot_DEA[!is.na(label1), ]
prot_DEA[!is.na(label3), ]

#load Uniprot identifiers
wtbl <- read.tab("Data/pullDown/uniprot-human-filtered-organism Homo+sapiens+(Human)+[9606] .tab")

#join relevant information from DEA and uniprot idenfier in master table
#PROTEIN
prot_tbl <- prot_DEA[,c(6,7,8,12)]
joined_prot<- wtbl %>% 
  filter(Status == "reviewed") %>%
  mutate(gname = str_extract(`Gene.names`, "[:graph:]+")) %>%
  select(Entry, gname) %>%
  right_join(prot_tbl)

hmitocarta_tbl<-as.data.frame(hmitocarta_tbl)
allmitocarta_tbl<-as.data.frame(allmitocarta_tbl)
# as an index to A$C
# A$C[A$C %in% B$C]

#for all genes in prot_dea
shared<-hmitocarta_tbl$UniProt[hmitocarta_tbl$UniProt %in% joined_prot$Entry] #563 entries of total 4395 -> 12.81%
sharedall <- allmitocarta_tbl$UniProt[allmitocarta_tbl$UniProt %in% joined_prot$Entry] #4302 entries of 4395 -> 97.88%

#for all genes with p<0.01 & ES>|0.2|
prot_dea_sig <- as.data.table(joined_prot)

prot_dea_sig[, mito := "Other"]
prot_dea_sig[which(prot_dea_sig$Entry %in% hmitocarta_tbl$UniProt), mito := "Mitochondrial"]
prot_dea_sig$mito <- factor(prot_dea_sig$mito)
summary(prot_dea_sig$mito)
prot_dea_sig[, status := NA_character_]

# â‰¤

pval_cutoff <- 0.01
prot_dea_sig[!is.na(pval) & !is.na(betas) & pval <= pval_cutoff & abs(betas) > 0.2, status := "p<0.01, ES>|0.2|"]
prot_dea_sig[!is.na(pval) & !is.na(betas) & pval > pval_cutoff & abs(betas) < 0.2, status := "Not sign."]
prot_dea_sig[!is.na(pval) & !is.na(betas) & pval <= pval_cutoff & abs(betas) < 0.2, status := "p<0.01"]
prot_dea_sig[!is.na(pval) & !is.na(betas) & pval > pval_cutoff & abs(betas) > 0.2, status := "ES>|0.2|"]
prot_dea_sig$status <- factor(prot_dea_sig$status, levels = c("Not sign.", "p<0.01", "ES>|0.2|", "p<0.01, ES>|0.2|"))
summary(prot_dea_sig$status)




ggplot(prot_dea_sig, aes(x = status, fill = mito)) +
  geom_bar(position = "fill", alpha = 0.6) +
  labs(fill = "Protein\nlocation") +
  ylab("Proportion") +
  ggtitle("Proteomics DEA outcome") +
  scale_fill_jco() +
  theme_pubr() +
  rotate_x_text(angle = 45) +
  guides(fill = guide_legend(nrow = 2)) +
  theme(plot.title = element_text(size =12), legend.position = "bottom", axis.title.x = element_blank())

ggsave(paste(fig_path,"ProteomicsDEAMitolocation.png", sep = ""), width = 2.7, height = 4)
dev.off()
ggsave(paste(fig_path_pdf, "Fig3/","ProteomicsDEAMitolocation.pdf", sep = ""), device = "pdf", width = 2.7, height = 4, bg = "white")
dev.off()



ggplot(prot_dea_sig, aes(x = status, fill = mito)) +
  geom_bar(position = "fill", alpha = 0.6) +
  labs(fill = "Protein location") +
  ylab("Proportion") +
  ggtitle("Proteomics DEA outcome") +
  scale_fill_jco() +
  coord_flip() +
  theme_pubr() +
  rotate_x_text(angle = 0) +
  guides(fill = guide_legend(nrow = 2)) +
  theme(plot.title = element_text(size =12), legend.position = "right", axis.title.y = element_blank(), axis.ticks.y = element_blank())

ggsave(paste(fig_path_pdf, "Fig3/","ProteomicsDEAMitolocationHORIZON.pdf", sep = ""), device = "pdf", width = 6, height = 3, bg = "white")
dev.off()

