# MMAomics

This repository provides all R scripts used to analyse multi-omics and phenotype datasets in Forny et al. 2022. A [pre-print](
https://www.medrxiv.org/content/10.1101/2022.01.27.22269972) of the publication has been deposited on the medRxiv server.


The code is structured according to the figures as presented in the paper.

## Table of contents
### Figures
1. Multi-facetted omics view enabled a molecular diagnosis.
2. Phenomics analysis reveals two main surrogate markers of disease severity.
3. Untargeted integration of omics data layers highlights the TCA cycle and associated pathways as well as oxidative phosphorylation gene sets to be dysregulated in MMA.
4. Transcript-protein and protein-protein correlation analyses reveal coordinated relationships between MMUT and TCA genes and proteins but not their isoforms.
5. Polar metabolomics in selected patient fibroblasts and glutamine tracing studies in CRISPR/Cas9 KO 293T cells highlight differential glutamine anaplerosis.
6. MMUT interacts physically with GLUD1, DLST, and GOT2 as demonstrated by FLAG-tag pull-down.

### Extended data Figures
1. Historic context of sample collection and quality control measurements of multi-omics data.
2. Biochemical assessment of MMUT activity and propionate incorporation activity supports diagnosis of affected individuals.
3. Expression outlier analysis reveals causative genes in specific disease samples.
4. The clinical severity score and propionate incorporation activity are associated with several phenotypic traits.
5. Global computational approaches to transcriptomics and proteotyping datasets failed to stratify samples into disease and non-disease groups.
6. Transcriptomics analysis of mouse brain revealed sample clustering according to genotype.
7. Significantly dysregulated proteins were enriched for for mitochondrial localization.
8. Transcript-protein and protein-protein correlation analysis illustrates coordinated regulation of MMUT with most TCA transcripts and proteins.
9. Metabolomics investigation of a subset of patient cell lines and glutamine flux studies in 293T CRISPR-KO cells.
10. Validation of CRISPR knock-out 293T cell lines.
11. Stable isotope labeling of glutamine shows preferential reductive cycling of glutamine derived carbons in MMUT deficiency.
